FeatureUpsell story: Allow toggling upsell state within the story

d-claassen opened the pull request:
## Context
<!--
What do we want to achieve with this PR? Why did we write this code?
-->

* Allow anyone interested in our `FeatureUpsell` to switch between the upsell and non-upsell state.

## Summary

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another add-on, start your changelog item with the name of that add-on's repo between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the changelog items is meant for the changelog of a javascript package, specify between square brackets in which package changelog the item should be included, for example: * [@yoast/components] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/add-ons, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* [@yoast/ui-library] Allows anyone interested in our `FeatureUpsell` to switch between the upsell and non-upsell state.

## Relevant technical choices:

* Start with the initial `args.shouldUpsell` to allow switching from the Storybook controls.
* Adding an `onClick` to an `<a>` tag, so that callback needs `event.preventDefault` etc.

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Pull this PR,
* Start the storybook with `$ yarn workspace @yoast/ui-library storybook`
* Navigate to the "Feature upsell" component stories,
* Try clicking the upsell buttons in any of the stories,
* Expect to see the content unlocked,
* Play around with the options in the controls and ensure all still work correctly and render the `FeatureUpsell` component as expected.

#### Relevant test scenarios
* [x] Changes should be tested with the browser console open
* [ ] Changes should be tested on different posts/pages/taxonomies/custom post types/custom taxonomies
* [ ] Changes should be tested on different editors (Block/Classic/Elementor/other)
* [ ] Changes should be tested on different browsers
* [ ] Changes should be tested on multisite
<!--
If you have checked any of the above cases, please add some context about the reason, what to check in the console,
which type/editor/browser should be tested in particular, multisite with subfolders or subdomains, etc.
-->

### Test instructions for QA when the code is in the RC
<!--
Sometimes some steps from the test instructions for the acceptance test aren't relevant anymore once the code has been merged or the feature is complete. If that is the case, do not check the checkbox below.
QA is our Quality Assurance team. The RC is the release candidate zip that is tested before a release
-->

* [ ] QA should use the same steps as above.

<!--
If the above checkbox has not been checked, write down all steps QA should take to test this PR, not only the difference with the acceptance test steps. If QA should use the test instructions specified on the epic, paste a link to the relevant comment on the epic.
-->
QA can test this PR by following these steps:

*

## Impact check
<!--
Sometimes PRs have a bigger impact than is suggested in the user-facing changes. In such cases,
additional (regression) testing might be necessary. To make it clear what parts might need additional testing, please outline which parts of the plugin have been impacted by this PR.
-->
This PR affects the following parts of the plugin, which may require extra testing:

* Only affects the Storybook of the UI library.

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Other environments

* [ ] This PR also affects Shopify. I have added a changelog entry starting with `[shopify-seo]`, added test instructions for Shopify and attached the `Shopify` label to this PR.

## Documentation

* [ ] I have written documentation for this change.

## Quality assurance

* [ ] I have tested this code to the best of my abilities.
* [ ] During testing, I had activated all plugins Yoast SEO provides integrations for.
* [ ] I have added unit tests to verify the code works as intended.
* [ ] If any part of the code is behind a feature flag, my test instructions also cover cases where the feature flag is switched off.
* [ ] I have written this PR in accordance with my team's definition of done.

## Innovation

* [ ] No innovation project is applicable for this PR.
* [ ] This PR falls under an innovation project. I have attached the `innovation` label and noted the work hours.

Fixes https://github.com/Yoast/ui-library-storybook/issues/12

vraja-pro commented:
I think adding a story that is the same as the factory component is maybe redundant?
Maybe adding a story with active and locked content side by side and then you don't need the button for returning to the upsell state? You could also adjust the text to be "Content that would be grayscaled" and "Active content".
I also think the blue is a bit out of place with the package UI colors, maybe it's better to use a more subtle color.

d-claassen commented:
Hey @vraja-pro, I found this old PR of mine again and decided to update it.

> I think adding a story that is the same as the factory component is maybe redundant?

I agree. The component has a bit of a silly default. I've enforced the default variant in the factory story now in
[Enforce default variant in the factory story](https://github.com/Yoast/wordpress-seo/pull/20263/commits/f51054e6573cbc014010c9752203eb16724be846).

> Maybe adding a story with active and locked content side by side and then you don't need the button for returning to the upsell state?

I personally prefer the button to give that interactive feeling. We have similar buttons to toggle toasts and modals.

> You could also adjust the text to be "Content that would be grayscaled" and "Active content". I also think the blue is a bit out of place with the package UI colors, maybe it's better to use a more subtle color.

You're right. I've added additional explanatory copy behind the upsell to better tell (and show) what's happening in [Clarify the behaviour op the locked content in a FeatureUpsell story](https://github.com/Yoast/wordpress-seo/pull/20263/commits/80919f541af9c18cb25083c63875df90a16810cf). And I've removed the strange blue background color in preference of the primary button, which demonstrates the same effect in [Realistic grayscale example](https://github.com/Yoast/wordpress-seo/pull/20263/commits/89aeb9212c9653e81c6772802973ff0f18e69d21).

vraja-pro commented:
CR & AC ✅
